### PR TITLE
Add libhdbpp recipe

### PR DESCRIPTION
**Pull request by beenje**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by beenje** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/libhdbpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://gitlab.com/tango-controls/hdbpp/libhdbpp/-/archive/v{{ version }}/libhdbpp-v{{ version }}.tar.gz
+  sha256: 53e1647d0106787adaed5d9e82c3eb53c5d5b1a4892387b1e52a183fadde4b93
+
+build:
+  number: 0
+  skip: true  # [not linux]
```
Comment (carterbox) [inline recipes/libhdbpp/meta.yaml:14]: Looks like upstream supports linux and windows.

```suggestion
  skip: true  # [osx]
```

File: recipes/libhdbpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+requirements:
+  build:
+    - cmake
+    - make
+    - {{ compiler('cxx') }}
+    - libtool
+    - pkg-config
+  host:
+    - cppzmq
+    - cpptango
```
Comment (carterbox) [inline recipes/libhdbpp/meta.yaml:25]: The [logs shows](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=521716&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1341) that you are probably statically linking against cpptango.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/libhdbpp/meta.yaml (hunk 2 above)
Comment (beenje) [reply recipes/libhdbpp/meta.yaml:25]: The `cpptango` package only contains a shared library and some headers.
Looks like `libhdbpp` is only creating an interface library: https://gitlab.com/tango-controls/hdbpp/libhdbpp/-/blob/master/cmake/FindTango.cmake#L15, so it's not linking against it.

`cpptango` is needed to compile this library and it defines a `run_exports`. Should I ignore this `run_exports`?
`libhdbpp` is an interface library and will be used to compile other packages.

**Comment by beenje** [comment]
@bourtemb do you have additional comments?

File: recipes/libhdbpp/meta.yaml (hunk 2 above)
Comment (carterbox) [reply recipes/libhdbpp/meta.yaml:25]: If there are no linkage/symbols from libhdb++.so to/in libcpptango.so, then it is not a runtime dependency of this library and the run_export should be ignored.

I'm not sure what you mean by interface library. If downstream packages need to link against cpptango, they should explicitly add it to their recipe host requirements.

File: recipes/libhdbpp/meta.yaml (hunk 1 above)
Comment (carterbox) [reply recipes/libhdbpp/meta.yaml:14]: pkg-config is available for Windows? CMake does not require platform specific directions. Does pkg-config require platform specific directions? I don't know; I've never used pkg-config.

File: recipes/libhdbpp/meta.yaml (hunk 2 above)
Comment (beenje) [reply recipes/libhdbpp/meta.yaml:28]: OK, thanks. `run_exports` ignored.

File: recipes/libhdbpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  url: https://gitlab.com/tango-controls/hdbpp/libhdbpp/-/archive/v{{ version }}/libhdbpp-v{{ version }}.tar.gz
+  sha256: 53e1647d0106787adaed5d9e82c3eb53c5d5b1a4892387b1e52a183fadde4b93
+
+build:
+  number: 0
+  skip: true  # [not linux]
+  # Library doesn't link against cpptango (only headers used)
+  ignore_run_exports:
+    - cpptango
+
```
Comment (carterbox) [inline recipes/libhdbpp/meta.yaml:18]:
```suggestion
  run_exports:
    - {{ pin_subpackage('libhdbpp') }}

```
People that build against this library may not need cpptango, but they definitely need this library? I assume this project uses SEMVER?

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/libhdbpp/meta.yaml (hunk 3 above)
Comment (beenje) [reply recipes/libhdbpp/meta.yaml:20]: Good point. Thanks! Added.

**Pull request merged**